lib/licensed/sources/gradle.rb

```diff
@@ -4,132 +4,164 @@
           name = "#{package["group"]}:#{package["name"]}"
           Dependency.new(
             name: name,
             version: package["version"],
-            path: File.join(@config.pwd, GRADLE_LICENSES_PATH, name),
+            path: config.pwd,
+            executable: gradle_executable,
+            configurations: configurations,
             metadata: {
               "type"     => Gradle.type
             }
           )
         end
       end
 
-      def with_latest_licenses
-        download_and_cache_licenses do
-          yield
-        end
-      end
-
       private
 
-      def download_licenses_from_csv
-        path = File.join(@config.pwd, GRADLE_LICENSES_PATH)
-        downloaded_licenses = {}
-        CSV.foreach(File.join(path, GRADLE_LICENSES_CSV_NAME), headers: true) do |row|
-          url = row["moduleLicenseUrl"]
-          artifact, _, version = row["artifact"].rpartition(":")
-          file_path = File.join(path, artifact, "LICENSE")
-          FileUtils.mkdir_p(File.join(path, artifact))
-          if cache_path = downloaded_licenses[url]
-            FileUtils.cp(cache_path, file_path)
-          else
-            uri = URI(row["moduleLicenseUrl"])
-            File.write(file_path, Net::HTTP.get(uri))
-            downloaded_licenses[url] = file_path
-          end
+      def gradle_executable
+        return @gradle_executable if defined?(@gradle_executable)
+        @gradle_executable = begin
+          gradlew = File.join(config.pwd, "gradlew")
+          return gradlew if File.executable?(gradlew)
+          "gradle" if Licensed::Shell.tool_available?("gradle")
         end
       end
 
-      def download_and_cache_licenses
-        gradle_command("generateLicenseReport")
-        download_licenses_from_csv
-        yield
-        FileUtils.rm_rf(File.join(@config.pwd, GRADLE_LICENSES_PATH))
-      end
-
-      # Returns the output from running `npm list` to get package metadata
-      def package_metadata_command
-        gradle_command("printDependencies")
-      end
-
-      def gradle_command(*args)
-        Tempfile.open(["license", ".gradle"], @config.pwd) do |f|
-          f.write gradle_file
-          f.close
-          Licensed::Shell.execute(executable, "-q", "-b", f.path, *args)
+      # Returns the configurations to include in license generation.
+      # Defaults to ["runtime", "runtimeClasspath"]
+      def configurations
+        @configurations ||= begin
+          if configurations = config.dig("gradle", "configurations")
+            Array(configurations)
+          else
+            DEFAULT_CONFIGURATIONS
+          end
         end
       end
 
-      def executable
-        if File.executable?(File.join(@config.pwd, "gradlew"))
-          "./gradlew"
-        else
-          "gradle"
+      def self.gradle_command(*args, path:, executable:, configurations:)
+        Dir.chdir(path) do
+          Tempfile.create(["license-", ".gradle"], path) do |f|
```

Comment:
switched this back to `.create` after noticing that `.open` doesn't immediately remove the created file.  From the [docs](https://ruby-doc.org/stdlib-2.3.0/libdoc/tempfile/rdoc/Tempfile.html), it seems like this is the major difference between `.create` and `.open` when a block is given 🤷‍♂️